src/foam/lib/json/FObjectParser.java

```diff
@@ -60,9 +60,27 @@ public PStream parse(PStream ps, ParserContext x) {
             if ( ps1 != null ) {
               try {
                 c = Class.forName(ps1.value().toString());
-              }catch (ClassNotFoundException t) {
-                x.set("error", t.getMessage() + " " + ps1.value().toString());
-                throw new RuntimeException(t);
+              } catch (ClassNotFoundException t) {
```

Comment:
It looks like this code is still the same. No need to do a Class.forName() when you can just use the string version of the classname on line 67 below.

Reply:
@kgrgreer I had to keep the class instance (i.e., return value from Class.forName(...)) because there were codes that were using this class instance to make the parser to work in FObjectParser (if omitted a lot of parser errors will be introduce). Instead, I reorganized the code in FObjectParser so that when the class (instance) does not exist, the string version is used to create an object.

Reply:
Also the string version of create is limited in what it can do because it doesn't have a reference to class; it can only check if the class has a custom factory. On the other hand, the class version of create can do other things like checking if the class is enum, abstract or getting XArgsFactory axiom, ... . If these checks are omitted, the parser won't work as expected. For this reason, I couldn't remove the class version of create and had to only use the string version as a "fallback".